platform/linux-generic/odp_ishm.c

```diff
@@ -2174,3 +2174,69 @@ void _odp_ishm_print(int block_index)
 
 	odp_spinlock_unlock(&ishm_tbl->lock);
 }
+
+int32_t odp_system_meminfo(odp_system_meminfo_t *info, odp_system_memblock_t memblock[],
+			   int32_t max_num)
+{
+	ishm_block_t *block;
+	int name_len, proc_index;
+	int32_t i;
+	uintptr_t addr;
+	uint64_t len, lost, page_size;
+	uint64_t lost_total = 0;
+	uint64_t len_total = 0;
+	int32_t num = 0;
+	const uint64_t huge_sz = odp_sys_huge_page_size();
+	const uint64_t normal_sz = odp_sys_page_size();
+
+	odp_spinlock_lock(&ishm_tbl->lock);
+	procsync();
+
+	for (i = 0; i < ISHM_MAX_NB_BLOCKS; i++) {
+		block = &ishm_tbl->block[i];
+
+		len = block->len;
+		if (len == 0)
+			continue;
+
+		lost = len - block->user_len;
+
+		if (num < max_num) {
+			name_len = strlen(block->name);
+			if (name_len >= ODP_SYSTEM_MEMBLOCK_NAME_LEN)
+				name_len = ODP_SYSTEM_MEMBLOCK_NAME_LEN - 1;
+
+			memcpy(memblock[num].name, block->name, name_len);
+			memblock[num].name[name_len] = 0;
+
+			addr = 0;
+			proc_index = procfind_block(i);
+			if (proc_index >= 0)
+				addr = (uintptr_t)ishm_proctable->entry[proc_index].start;
```

Comment:
I guess in theory `addr` could be left as zero if `procfind_block()` fails. Is this OK or should something be said in spec about failure values for `odp_system_memblock_t:addr`?

Reply:
I think this should not fail in normal case (we are holding lock while going through SHM blocks). But even if it would, then it might be better to continue than stop here, since the API is for debugging SHM mappings.